python/ray/serve/config.py

```diff
@@ -55,6 +59,24 @@ def set_max_queries_by_mode(cls, v, values):  # noqa 805
                 raise ValueError("max_concurrent_queries must be >= 0")
         return v
 
+    def to_proto_bytes(self):
+        data = self.dict()
+        if "user_config" in data:
+            data["user_config"] = pickle.dumps(data["user_config"])
+        return BackendConfigProto(
+            is_cross_language=False,
+            backend_language=BackendLanguage.PYTHON,
+            **data,
+        ).SerializeToString()
+
+    @classmethod
+    def from_proto_bytes(cls, proto_bytes: bytes):
+        proto = BackendConfigProto.FromString(proto_bytes)
+        data = MessageToDict(proto, preserving_proto_field_name=True)
+        if "user_config" in data:
+            data["user_config"] = pickle.loads(proto.user_config)
+        return cls(**data)
+
```

Comment:
Can we add a check that this never gets serialized using pickle to avoid mistakes? Could just throw an exception in `__reduce__`.